Applications/spire/include/spire/ui/icon_button.hpp

```diff
@@ -20,15 +20,16 @@ namespace Spire {
         \param icon The icon to show.
         \param parent The parent QWidget to the IconButton.
       */
-      IconButton(QImage icon, QWidget* parent = nullptr);
+      IconButton(const QImage& icon, QWidget* parent = nullptr);
```

Comment:
Mark ```explicit```.

Reply:
Why is the pass by value/move semantics eliminated from this class?

Reply:
I removed it because the ToggleButton class is keeping instances of the images so that it can pass them to IconButton's set_icon() function when the state changes.

Reply:
One class' API should not influence another class' API. They are two independent classes. Unless IconButton has a specific use for passing by reference then it's preferable to use value/move semantics.